packages/@aws-cdk/core/lib/custom-resource-provider.ts

```diff
@@ -0,0 +1,166 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { AssetStaging } from './asset-staging';
+import { FileAssetPackaging } from './assets';
+import { CfnResource } from './cfn-resource';
+import { Construct } from './construct-compat';
+import { Duration } from './duration';
+import { Size } from './size';
+import { Stack } from './stack';
+import { Token } from './token';
+
+const ENTRYPOINT_FILENAME = '__entrypoint__';
+const ENTRYPOINT_HANDLER = path.join(__dirname, 'custom-resource-provider-handler.js');
+
+/**
+ * Initialization properties for `CustomResourceProvider`.
+ *
+ * @experimental
+ */
+export interface CustomResourceProviderProps {
+  /**
+   * A local file system directory with the provider's code. The code will be
+   * bundled into a zip asset and wired to the provider's AWS Lambda function.
+   */
+  readonly codeDirectory: string;
+
+  /**
+   * The AWS Lambda runtime and version to use for the provider.
+   */
+  readonly runtime: CustomResourceProviderRuntime;
+
+  /**
+   * A set of IAM policy statements to include in the inline policy of the
+   * provider's lambda function.
+   *
+   * @default - no additional inline policy
+   */
+  readonly policyStatements?: any[];
```

Comment:
Some examples would help decipher this any type

Reply:
added